api/v1beta1/awscluster_types.go

```diff
@@ -198,6 +205,23 @@ type AWSClusterStatus struct {
 	Conditions     clusterv1.Conditions     `json:"conditions,omitempty"`
 }
 
+type S3Bucket struct {
+	// Create controls if S3 Bucket should be created as part of cluster provisioning.
+	Create bool `json:"create,omitempty"`
+
+	// ControlPlaneIAMInstanceProfile is a name of the IAMInstanceProfile, which will be allowed
+	// to read control-plane node bootstrap data from S3 Bucket.
+	ControlPlaneIAMInstanceProfile string `json:"controlPlaneIAMInstanceProfile,omitempty"`
+
+	// NodesIAMInstanceProfiles is a list of IAM instance profiles, which will be allowed to read
+	// worker nodes bootstrap data from S3 Bucket.
+	NodesIAMInstanceProfiles []string `json:"nodesIAMInstanceProfiles,omitempty"`
+
+	// Name defines name of S3 Bucket to be created. Must start with 'cluster-api-provider-aws-'
+	// by default.
+	Name string `json:"name,omitempty"`
```

Comment:
Does it make sense to split the validation logic between type declaration and webhook code? It is desired to leverage kubebuilder validation as much as possible? How are kubebuilder validation rules tested?

Reply:
I saw your comment about IP parsing, that is a strange bucket naming rule from AWS.

Since that validation logic is most likely never to be changed, would prefer to simplify the webhook code and offload to OpenAPI as much as possible, but leaving it up to you.

Reply:
I see that unit tests already use webhooks validation so it make sense to move it to kubebuilder annotation indeed. Done so marking as resolved.